### PR TITLE
113 front development

### LINKED ISSUE #147: Add a Front for dynamic params
**Short description:**

Add a front style to our dynamic params

### LINKED ISSUE #163: [BUG] AREA description display null instead of none
**Describe the bug**
When a description is empty, it displays null 'instead' of 'none' or something else

### LINKED ISSUE #164: [BUG] Name of area and description pass in overflow
**Describe the bug**
Name of area pass in overflow

**To Reproduce**
```
Row(
  children: const [
    Expanded(
      child: Text(
        'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.',
        style: TextStyle(fontSize: 58),
        softWrap: false,
        maxLines: 1,
        overflow: TextOverflow.ellipsis, // new
      ),
    ),
  ],
)
```

### LINKED ISSUE #165: [BUG] Add a max size to display area in home page
**Describe the bug**
Clear Area display on homepage

### LINKED ISSUE #169: Add way to personalize area
**Short description:**

Add a way to personalize area in Flutter app.

**Describe the solution you'd like**
Store AREA icon and color inside the database.
Add it in Update/Create screen and app call.

### LINKED ISSUE #170: Display Actions and reaction only if toggle button is true
**Short description:**
A clear and concise description of what the problem is. Ex. Add the action .. from the service ... .

**Describe the solution you'd like**
Ex: Add the action ... from the service ... to the possibles action choices. It should be a red square. But it should not be able to be linked with the reaction ... from the service ... .

### PR DESCRIPTION
**Pull request by TomDUVAL-MAHE**
# Purpose

Front development

Fixes #170
Fixes #169
Fixes #165
Fixes #164
Fixes #163
Fixes #147

## Added features

Front beautifulage

## Updated features

Too many

## Fixed feature

Too many

## Removed features

Too many

If you want to test it, ask me :wink:

### CODE REVIEW

File: application/app_code/lib/flutter_objects/action_data.dart [hunk 1]
```diff
@@ -74,36 +82,36 @@ class ActionData {
...
-                      description,
-                      style: TextStyle(
-                          color: isEnable ? Colors.green : Colors.red),
-                    ), // Change when icon are in DB
-                  ]),
-            ],
-          ),
-        ]));
-    if (mode == true) {
+  Widget displayActionModificationView(Function? update) {
```
Comment (LucasTesnier) [inline application/app_code/lib/flutter_objects/action_data.dart:86]: missing doc

File: application/app_code/lib/flutter_objects/reaction_data.dart [hunk 2]
```diff
@@ -104,6 +111,20 @@ class ReactionData {
     );
   }
 
+  Widget displayReactionDescription() {
```
Comment (LucasTesnier) [inline application/app_code/lib/flutter_objects/reaction_data.dart:115]: missing doc

File: application/app_code/lib/pages/create_area/create_area_state.dart [hunk 3]
```diff
@@ -10,19 +10,156 @@ import '../../flutter_objects/action_data.dart';
...
 import '../../flutter_objects/service_data.dart';
+import '../../material_lib_functions/material_functions.dart';
 import '../../network/informations.dart';
 import '../home/home_functional.dart';
 
 class CreateAreaPageState extends State<CreateAreaPage> {
   /// Setting of the action set ?
   bool actionSetting = false;
 
+  static const List<Widget> listActionCondition = <Widget>[
```
Comment (LucasTesnier) [inline application/app_code/lib/pages/create_area/create_area_state.dart:22]: missing doc

File: application/app_code/lib/pages/create_area/create_area_state.dart [hunk 4]
```diff
@@ -10,19 +10,156 @@ import '../../flutter_objects/action_data.dart';
...
+    Text('Or'),
+    Text('And')
+  ];
+
+  static const List<Widget> listPrimaryColor = <Widget>[
```
Comment (LucasTesnier) [inline application/app_code/lib/pages/create_area/create_area_state.dart:28]: same

File: application/app_code/lib/pages/create_area/create_area_state.dart [hunk 5]
```diff
@@ -10,19 +10,156 @@ import '../../flutter_objects/action_data.dart';
...
+      Icons.color_lens,
+      color: Colors.amber,
+    ),
+    Icon(
+      Icons.color_lens,
+      color: Colors.black,
+    )
+  ];
+
+  static const List<Widget> listSecondaryColor = <Widget>[
```
Comment (LucasTesnier) [inline application/app_code/lib/pages/create_area/create_area_state.dart:52]: same

File: application/app_code/lib/pages/create_area/create_area_state.dart [hunk 6]
```diff
@@ -10,19 +10,156 @@ import '../../flutter_objects/action_data.dart';
...
+      Icons.color_lens_outlined,
+      color: Colors.amber,
+    ),
+    Icon(
+      Icons.color_lens_outlined,
+      color: Colors.black,
+    )
+  ];
+
+  static List<Widget> listIcon = <Widget>[
```
Comment (LucasTesnier) [inline application/app_code/lib/pages/create_area/create_area_state.dart:76]: same

File: application/app_code/lib/pages/create_area/create_area_state.dart [hunk 7]
```diff
@@ -10,19 +10,156 @@ import '../../flutter_objects/action_data.dart';
...
+      key: const Key('assets/icons/Area_Logo.png'),
+      width: 21,
+      height: 20,
+    ),
+  ];
+
   /// Creation of an Action state
   int _actionCreationState = 0;
 
+  bool isSelected = false;
```
Comment (LucasTesnier) [inline application/app_code/lib/pages/create_area/create_area_state.dart:125]: same

File: application/app_code/lib/pages/create_area/create_area_state.dart [hunk 8]
```diff
@@ -10,19 +10,156 @@ import '../../flutter_objects/action_data.dart';
...
+
+  final List<bool> _selectedActionCondition = createdArea != null
```
Comment (LucasTesnier) [inline application/app_code/lib/pages/create_area/create_area_state.dart:128]: same

File: application/app_code/lib/pages/create_area/create_area_state.dart [hunk 9]
```diff
@@ -10,19 +10,156 @@ import '../../flutter_objects/action_data.dart';
...
+      ? createdArea!.logicalGate == 'OR'
+          ? <bool>[true, false]
+          : <bool>[false, true]
+      : <bool>[true, false];
+
+  final List<bool> _selectedPrimaryColor = <bool>[
```
Comment (LucasTesnier) [inline application/app_code/lib/pages/create_area/create_area_state.dart:135]: same

File: application/app_code/lib/pages/create_area/create_area_state.dart [hunk 10]
```diff
@@ -10,19 +10,156 @@ import '../../flutter_objects/action_data.dart';
...
+    false,
+    false,
+    false,
+    false,
+    true
+  ];
+
+  final List<bool> _selectedSecondaryColor = <bool>[
```
Comment (LucasTesnier) [inline application/app_code/lib/pages/create_area/create_area_state.dart:144]: same

File: application/app_code/lib/pages/create_area/create_area_state.dart [hunk 11]
```diff
@@ -10,19 +10,156 @@ import '../../flutter_objects/action_data.dart';
...
+    false,
+    false,
+    false,
+    false,
+    true
+  ];
+
+  final List<bool> _selectedIcon = <bool>[
```
Comment (LucasTesnier) [inline application/app_code/lib/pages/create_area/create_area_state.dart:153]: same

File: application/app_code/lib/pages/home/home_state.dart [hunk 12]
```diff
@@ -24,6 +24,12 @@ class HomePageState extends State<HomePage> {
     updatePage = update;
   }
 
+  bool isDesktop(BuildContext context) =>
```
Comment (LucasTesnier) [inline application/app_code/lib/pages/home/home_state.dart:28]: missing doc

File: application/app_code/lib/pages/home/home_state.dart [hunk 13]
```diff
@@ -24,6 +24,12 @@ class HomePageState extends State<HomePage> {
...
+      MediaQuery.of(context).size.width >= 600;
+
+  bool isMobile(BuildContext context) =>
```
Comment (LucasTesnier) [inline application/app_code/lib/pages/home/home_state.dart:32]: same

**Review by LucasTesnier — CHANGES_REQUESTED** [review]
Just documentation otherwise NTR

**Review by ZQUEMA — DISMISSED** [review]
Back OK

**Review by LucasTesnier — APPROVED** [review]

**Review by ZQUEMA — APPROVED** [review]

**Pull request merged**